File: anmol JS/ac.js
var HUMIDITY_RATE = -2/3600; // -2% per hour
var TEMPERATURE_RATE = -10/3600; // -10C per hour
var VOLUME_AT_RATE = 100000;

var input;
function setup() {
	IoEClient.setup({
		type: "AC",
		states: [{
			name: "On",
			type: "bool",
			controllable: true
		}]
	});
	
	IoEClient.onInputReceive = function(data) {
		if ( data.length <= 0  )
			return;
		data = data.split(",");
		processData(parseInt(data[0]));
	};

	attachInterrupt(0, function() {
		processData(digitalRead(0)/1023);
	});

	processData(digitalRead(0)/1023);
	
	var VAR = getDeviceProperty(getName(), "VOLUME_AT_RATE");
	if(!VAR)
		setDeviceProperty(getName(), "VOLUME_AT_RATE", VOLUME_AT_RATE);
}

function processData(data) {
	input = data;
	if ( input > 0 )
		digitalWrite(5, HIGH);
	else
		digitalWrite(5, LOW);
	IoEClient.reportStates(input);
}

function loop()
{
	updateEnvironment();
	delay(1000);
}


function updateEnvironment()
{
	var VAR = parseFloat(getDeviceProperty(getName(), "VOLUME_AT_RATE"));
	if( VAR < 0 )
		VAR = VOLUME_AT_RATE;
	//Serial.println("VAR: " + VAR);
	//Serial.println("input: " + input);

	var humidity_rate = input*HUMIDITY_RATE * VAR / Environment.getVolume();
	var temperature_rate = input*TEMPERATURE_RATE * VAR / Environment.getVolume();
	Environment.setContribution("Humidity", humidity_rate);
	Environment.setContribution("Ambient Temperature", temperature_rate);
	
	//Serial.println("T rate: " + temperature_rate);
}

File: anmol py/ac.py
from gpio import *
from time import *
from ioeclient import *
from environment import *
from physical import *

HUMIDITY_RATE = -2./3600; # -2% per hour
TEMPERATURE_RATE = -10./3600; # -10C per hour
VOLUME_AT_RATE = 100000.

myInput = 0


    
def onInputReceiveDone(data):
    if  len(data) <= 0:
        return
    data = data.split(",")
    processData(int(data[0]))
        

def detect():
    processData(digitalRead(0)/1023)
  
def setup():
    IoEClient.setup({
        "type": "AC",
        "states": [{
            "name": "On",
            "type": "bool",
            "controllable": True
        }]
    })
    IoEClient.onInputReceive(onInputReceiveDone)
    add_event_detect(0, detect)
    detect()
    
    VAR = getDeviceProperty(getName(), "VOLUME_AT_RATE")
    if not VAR:
    	setDeviceProperty(getName(), "VOLUME_AT_RATE", VOLUME_AT_RATE)



def processData(data):
    global myInput
    myInput = data
    if  myInput > 0 :
        digitalWrite(5, HIGH)
    else :
        digitalWrite(5, LOW)
    IoEClient.reportStates(myInput)


def updateEnvironment():
    VAR = float(getDeviceProperty(getName(), "VOLUME_AT_RATE"))
    if VAR < 0:
    	VAR = 0

    humidity_rate = float(myInput*HUMIDITY_RATE*VAR) / Environment.getVolume()
    temperature_rate = float(myInput*TEMPERATURE_RATE*VAR) / Environment.getVolume()
    Environment.setContribution("Humidity", humidity_rate, 0, True) 
    Environment.setContribution("Ambient Temperature", temperature_rate, -1000, True)
    #print temperature_rate, VAR

def main():
    setup()
    while True:
        updateEnvironment()
        delay(1000)
        
if __name__ == "__main__":
    main()